src/MOI_wrapper/MOI_wrapper.jl

```diff
@@ -126,367 +171,236 @@ function MOI.is_empty(model::Optimizer)
+        variable_upper,
+        c,
+        matrix.lower,
+        matrix.upper,
+    )
+    sense = MOI.get(src, MOI.ObjectiveSense())
+    if sense == MOI.MIN_SENSE
+        Cbc_setObjSense(dest, 1)
+    elseif sense == MOI.MAX_SENSE
+        Cbc_setObjSense(dest, -1)
+    else
+        @assert sense == MOI.FEASIBILITY_SENSE
+        Cbc_setObjSense(dest, 0)
+    end
+    Cbc_setInteger.(dest, binaries)
+    attr = MOI.ListOfConstraintIndices{MOI.VariableIndex,MOI.Integer}()
+    for ci in MOI.get(src, attr)
+        Cbc_setInteger(dest, Cint(ci.value - 1))
+    end
+    any_sos = false
+    for (S, type) in ((MOI.SOS1{Float64}, 1), (MOI.SOS2{Float64}, 2))
+        starts, indices, weights = Cint[], Cint[], Float64[]
+        attr = MOI.ListOfConstraintIndices{MOI.VectorOfVariables,S}()
+        for ci in MOI.get(src, attr)
+            any_sos = true
+            push!(starts, Cint(length(weights)))
+            f = MOI.get(src, MOI.ConstraintFunction(), ci)
+            for x in f.variables
+                push!(indices, Cint(x.value - 1))
+            end
+            s = MOI.get(src, MOI.ConstraintSet(), ci)
+            append!(weights, s.weights)
+        end
+        N = Cint(length(starts))
+        if N > 0
+            push!(starts, length(weights))
+            Cbc_addSOS(dest, N, starts, indices, weights, Cint(type))
+        end
+    end
+    if any_sos && Cbc_getNumIntegers(dest) == 0
+        @warn(
+            "There are known correctness issues using Cbc with SOS " *
+            "constraints and no binary variables.",
+        )
+    end
+    return _index_map(src)
 end
 
-function _load_constraint(
-    ::MOI.ConstraintIndex,
-    model::_CbcModelFormat,
-    mapping::MOI.IndexMap,
-    func::MOI.VariableIndex,
-    ::MOI.Integer,
+function MOI.copy_to(
+    dest::Optimizer,
+    src::MOI.Utilities.UniversalFallback{OptimizerCache},
 )
-    push!(model.integer, _column_value(mapping, func) - 1)
-    return
+    MOI.Utilities.throw_unsupported(src)
+    return MOI.copy_to(dest, src.model)
 end
 
-function _load_constraints(
-    model::_CbcModelFormat,
-    src::MOI.ModelLike,
-    mapping::MOI.IndexMap,
-    F::Type{<:MOI.AbstractFunction},
-    S::Type{<:MOI.AbstractSet},
-)
-    for index in MOI.get(src, MOI.ListOfConstraintIndices{F,S}())
-        _load_constraint(
-            index,
-            model,
-            mapping,
-            MOI.get(src, MOI.ConstraintFunction(), index),
-            MOI.get(src, MOI.ConstraintSet(), index),
-        )
+function MOI.copy_to(dest::Optimizer, src::MOI.ModelLike)
+    cache = MOI.default_cache(dest, Float64)
+    src_cache = MOI.copy_to(cache, src)
+    cache_dest = MOI.copy_to(dest, cache)
+    index_map = MOI.IndexMap()
+    for (src_x, cache_x) in src_cache.var_map
+        index_map[src_x] = cache_dest[cache_x]
     end
-    return
+    for (src_ci, cache_ci) in src_cache.con_map
+        index_map[src_ci] = cache_dest[cache_ci]
+    end
+    return index_map
```

Comment:
You could do `return src_cache` in case `cache_dest` is identity.

Reply:
It isn't: https://github.com/jump-dev/Cbc.jl/pull/189#discussion_r818092745